src/hive_mind_bridge/src/HiveMindBridgeImpl.cpp

```diff
@@ -80,14 +107,55 @@ void HiveMindBridgeImpl::inboundThread() {
             m_inboundQueue.push(message);
         }
     }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(250));
+}
+
+void HiveMindBridgeImpl::outboundThread() {
+    while (isTCPClientConnected()) {
+        if (!m_outboundQueue.empty()) {
+            OutboundRequestHandle handle = m_outboundQueue.front();
+
+            MessageDTO outboundMessage = handle.getMessage();
+            auto request = std::get_if<RequestDTO>(&outboundMessage.getMessage());
+
+            if (request) {
+                // verify if the front value has a corresponding response handle
+                auto search = m_inboundResponsesMap.find(request->getId());
+                if (search != m_inboundResponsesMap.end()) {
+                    // Received a response for this request so we delete the request
+                    // TODO add some retry logic in case the response was not ok
+                    m_outboundQueue.pop();
+                    m_inboundResponsesMap.erase(search);
+
+                    ROS_INFO("RECEIVED VALID RESPONSE");
+                } else {
+                    // Did not receive a response for this request. Was the request sent?
+                    if (handle.getState() == OutboundRequestState::READY) {
+                        m_serializer.serializeToStream(outboundMessage);
+                        handle.setState(OutboundRequestState::SENT);
+                    } else {
+                        // TODO add some retry logic after a timeout
```

Comment:
So right now, if the message is lost (eg we try to send it to a robot that is no longer in the swarm), we will never move on to the next message? Is this intended?

Reply:
Yes. This situation may arise in another context. Let's say that the bridge sends a Request, but that the response takes some time to come back. We do not want to drop/retry the message immediately. I'll add some timeout logic before dropping the message.

Reply:
Of course, this logic will only have its use when we define a retry logic.